File: Programação Orientada a Objetos/Atividades/Atividade Extra 2/ControleDeAudio CPP/ControleDeAudio.cpp


class ControleDeAudio
{
private:
    float volume;

public:
    ControleDeAudio()
    {
        volume = 2;
    }

    void AumentarVolume()
    {
        if (this->volume < 10) {
            this->volume++;
        }
    }

    void DiminuirVolume()
    {
        if (this->volume > 0)
        {
            this->volume--;
        }
    }

    float LerVolume(){
        return this->volume;
    }
};

File: Programacao Orientada a Objetos/Atividades/Atividade Extra 2/ControleDeAudio TS/src/ControleDeAudio.ts
class ControleDeAudio {
    private volume: number;

    public constructor() {
        this.volume = 2;
    }

    public AumentarVolume(): void {
        if (this.volume < 10) {
            this.volume++;
        }
    }

    public DiminuirVolume(): void {
        if (this.volume > 0) {
            this.volume--;
        }
    }

    public LerVolume(): number {
        return this.volume;
    }
}